editor/command_palette.cpp

```diff
@@ -0,0 +1,191 @@
+/*************************************************************************/
+/*  code_editor.h                                                        */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2021 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2021 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "editor/command_palette.h"
+#include "core/os/keyboard.h"
+#include "editor/editor_node.h"
+#include "scene/gui/tree.h"
+
+void CommandPalette::_update_search() {
+	print_line(command_search_box->get_text());
+}
+
+String CommandPalette::get_command_text() const {
+	const String cmd = command_search_box->get_text();
+	if (cmd[0] == '>') {
+		return cmd.substr(1, -1);
+	} else {
+		return cmd;
+	}
+}
+
+void CommandPalette::_update_command_search() {
+	List<String> *actions_list = memnew(List<String>);
```

Comment:
This would need to be freed with `memdelete(actions_list)`, but i think simply not making this a pointer is better.
You would use a `List<String> actions_list` instead. You would then call get_action_list with a reference instead `get_action_list(&actions_list)`